### PR TITLE
Update Dependencies and CI Scrips

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

Update Dependencies and CI Scrips

## Commits

- chore: update vm2.TestUtilities package version to 1.5.1 and enhance CI workflow with skip options
- chore: fix typo in merge or copy action description in CONVENTIONS.md
- chore: update vm2.TestUtilities package version to 1.5.1 and add Microsoft.Extensions.TimeProvider.Testing dependency

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [x] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: .github/CONVENTIONS.md [hunk 1]
```diff
@@ -93,13 +93,13 @@ The project owner is a non-native English speaker.
 - Files that are not shared and should be maintained separately in each repo are not marked
 - These conventions are implemented by the `diff-shared.sh` script, which is used to propagate changes from the source-of-truth files in the template content folder to the target repos. The script can be configured to specify which files are shared and how to handle changes to them. For even more nuanced and flexible control, it defines 6 different actions for handling changes to shared files:
   - **`ignore`**: do not update the target file, keep it as is
-  - **`merge_or_copy`**: ask the user to choose between ignoring, merging or copying the new content over the existing the new content.
-  - **`ask_to_merge`**: ask the user if they want to merge the new content with the existing file; if they choose not to merge, do not update the target file
+  - **`merge or copy`**: ask the user to choose between ignoring, merging or copying the new content over the existing new content.
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/CONVENTIONS.md]: ok

File: .github/workflows/CI.yaml [hunk 2]
```diff
@@ -173,4 +197,21 @@ jobs:
...
+              if [[ "${{ needs.prerun-ci.outputs.skip-benchmarks }}" == "true" ]]; then
+                  echo "⚠️  WARNING: Benchmarks were skipped. The CI workflow is marked as failed, but it was intentional to save time. Use this option only for testing non-benchmark CI jobs faster."
+                  exit_code=1
+              fi
+              if [[ "${{ needs.prerun-ci.outputs.skip-tests }}" == "true" ]]; then
+                  echo "⚠️  WARNING: Tests were skipped. The CI workflow is marked as failed, but it was intentional to save time. Use this option only for testing non-test CI jobs faster."
+                  exit_code=1
+              fi
+              if [[ "${{ needs.prerun-ci.outputs.skip-packages }}" == "true" ]]; then
+                  echo "⚠️  WARNING: Building packages were skipped. The CI workflow is marked as failed, but it was intentional to save time. Use this option only for testing non-package CI jobs faster."
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/workflows/CI.yaml]: ok
